app/build.gradle

```diff
@@ -1,23 +1,35 @@
 import com.android.ddmlib.DdmPreferences
 
+import java.security.InvalidParameterException
+
 apply plugin: 'com.android.application'
 
 android {
     compileSdkVersion 26
     defaultConfig {
-        applicationId "tech.pauly.find_a_pet"
+        applicationId "tech.pauly.findapet"
         minSdkVersion 21
         targetSdkVersion 26
         versionCode 1
         versionName "1.0"
-        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        testInstrumentationRunner "tech.pauly.findapet.EspressoTestRunner"
         DdmPreferences.setTimeOut(60000)
     }
     buildTypes {
         release {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
+
+        applicationVariants.all { variant ->
+            String environment = project.property("findapet.environment")
```

Comment:
Decided to go with environment switching through Gradle Properties as that's what I'm comfortable with. The only downside in my experience is they're not an enum but just a string so you can screw up. I attempt to curb this by using this check, and the build-time check below in `afterEvaluate` to ensure I don't run the app in espresso mode when I'm trying to run Espresso tests and vice versa, nor do fall victim to a typo when changing the field.

I could always use environment as a flavor instead, I've not in the past due to how long the variants list gets, but in this project that likely won't be an issue. Maybe I'll switch, but this works for now.